### PR TITLE
bug: Add some missing field in row group metadata: ordinal, total co…

### LINKED ISSUE #4610: bug: Parquet writer missing row group metadata fields such as `compressed_size`, `file offset`.
**Describe the bug**

Parquet writer missing row group metadata fields such as `compressed_size`, `file offset`.

**To Reproduce**

**Expected behavior**

It should fill these fields.

**Additional context**

These fields are important for other readers.

### PR DESCRIPTION
**Pull request by liurenjie1024**
# Which issue does this PR close?

Closes #4610.

# Rationale for this change
 

This is important for readers to split parquet into different parts, missing them may lead to bug.

# What changes are included in this PR?

# Are there any user-facing changes?

No.

### CODE REVIEW

File: parquet/src/file/metadata.rs [hunk 1]
```diff
@@ -277,6 +277,9 @@ pub struct RowGroupMetaData {
     sorting_columns: Option<Vec<SortingColumn>>,
     total_byte_size: i64,
     schema_descr: SchemaDescPtr,
+    // We can't infer from file offset of first column since there may empty columns in row group.
```
Comment (tustvold) [inline parquet/src/file/metadata.rs:280]: I think these should probably be i64 to match thrift

File: parquet/src/file/writer.rs [hunk 2]
```diff
@@ -184,6 +184,7 @@ impl<W: Write + Send> SerializedFileWriter<W> {
     pub fn next_row_group(&mut self) -> Result<SerializedRowGroupWriter<'_, W>> {
         self.assert_previous_writer_closed()?;
         self.row_group_index += 1;
+        let file_offset = self.buf.bytes_written();
```
Comment (tustvold) [inline parquet/src/file/writer.rs:187]:
```suggestion
        let file_offset = self.buf.bytes_written();
        let ordinal = self.row_group_index;
        self.row_group_index += 1;
```

File: parquet/src/file/metadata.rs (hunk 1 above)
Comment (tustvold) [reply parquet/src/file/metadata.rs:280]: I would tend to agree, however, for consistency with the rest of the APIs and with the parquet format I think this should be unsigned

File: parquet/src/file/writer.rs [hunk 3]
```diff
@@ -418,16 +425,22 @@ impl<'a, W: Write + Send> SerializedRowGroupWriter<'a, W> {
...
     /// - `buf` - the buffer to write data to
+    /// - `row_group_index` - row group index in this parquet file.
+    /// - `file_offset` - file offset of this row group in this parquet file.
     /// - `on_close` - an optional callback that will invoked on [`Self::close`]
     pub fn new(
         schema_descr: SchemaDescPtr,
         properties: WriterPropertiesPtr,
         buf: &'a mut TrackedWrite<W>,
+        row_group_index: usize,
+        file_offset: usize,
```
Comment (tustvold) [inline parquet/src/file/writer.rs:436]: Instead of passing this in, could we just read this from `buf`?

Comment (liurenjie1024) [reply parquet/src/file/writer.rs]: Good catch, we just need to pass in `row_group_index`.

**Comment by liurenjie1024** [comment]
cc @tustvold Any other comments?

**Comment by tustvold** [comment]
Thank you

**Pull request merged**